exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,2055 @@
+			AssetType:    assetType,
+			CurrencyPair: pair,
+			Exchange:     ok.Name,
+			Side:         side,
+			Timestamp:    response.Data[i].Timestamp,
+			TID:          response.Data[i].TradeID,
+			Price:        response.Data[i].Price,
+		}
+	}
+	return trade.AddTradesToBuffer(ok.Name, trades...)
+}
+
+// wsProcessOrders handles websocket order push data responses.
+func (ok *Okx) wsProcessOrders(respRaw []byte) error {
+	var response WsOrderResponse
+	var pair currency.Pair
+	var assetType asset.Item
+	var err error
+	err = json.Unmarshal(respRaw, &response)
+	if err != nil {
+		return err
+	}
+	assetType, err = ok.GetAssetTypeFromInstrumentType(response.Argument.InstrumentType)
+	if err != nil {
+		return err
+	}
+	for x := range response.Data {
+		orderType, err := order.StringToOrderType(response.Data[x].OrderType)
+		if err != nil {
+			ok.Websocket.DataHandler <- order.ClassificationError{
+				Exchange: ok.Name,
+				OrderID:  response.Data[x].OrderID,
+				Err:      err,
+			}
+		}
+		orderStatus, err := order.StringToOrderStatus(response.Data[x].State)
+		if err != nil {
+			ok.Websocket.DataHandler <- order.ClassificationError{
+				Exchange: ok.Name,
+				OrderID:  response.Data[x].OrderID,
+				Err:      err,
+			}
+		}
+		var a asset.Item
+		a, err = ok.GetAssetTypeFromInstrumentType(response.Argument.InstrumentType)
+		if err != nil {
+			ok.Websocket.DataHandler <- order.ClassificationError{
+				Exchange: ok.Name,
+				OrderID:  response.Data[x].OrderID,
+				Err:      err,
+			}
+			a = assetType
+		}
+		pair, err = ok.GetPairFromInstrumentID(response.Data[x].InstrumentID)
+		if err != nil {
+			return err
+		}
+		ok.Websocket.DataHandler <- &order.Detail{
+			Price:           response.Data[x].Price,
+			Amount:          response.Data[x].Size,
+			ExecutedAmount:  response.Data[x].LastFilledSize,
+			RemainingAmount: response.Data[x].AccumulatedFillSize - response.Data[x].LastFilledSize,
+			Exchange:        ok.Name,
+			OrderID:         response.Data[x].OrderID,
+			Type:            orderType,
+			Side:            response.Data[x].Side,
+			Status:          orderStatus,
+			AssetType:       a,
+			Date:            response.Data[x].CreationTime,
+			Pair:            pair,
+		}
+	}
+	return nil
+}
+
+// wsProcessCandles handler to get a list of candlestick messages.
+func (ok *Okx) wsProcessCandles(intermediate *wsIncomingData) error {
+	if intermediate == nil {
+		return errNilArgument
+	}
+	var response WSCandlestickResponse
+	if len(intermediate.Data) == 0 {
+		return errNoCandlestickDataFound
+	}
+	pair, err := ok.GetPairFromInstrumentID(intermediate.Argument.InstrumentID)
+	if err != nil {
+		return err
+	}
+	var a asset.Item
+	a, err = ok.GetAssetTypeFromInstrumentType(intermediate.Argument.InstrumentType)
+	if err != nil {
+		a = ok.GuessAssetTypeFromInstrumentID(intermediate.Argument.InstrumentID)
+	}
+	candleInterval := strings.TrimPrefix(intermediate.Argument.Channel, candle)
+	for i := range response.Data {
```

Comment:
Please verify this handling. Its not getting coverage from tests despite candlestick data being passed to it